### PR TITLE
Add some basic specs for the grammar

### PR DESCRIPTION
**Pull request by bitbutt**
I wrote some basic specs to serve as a sanity check for any grammar tweaks. All of these specs pass against the current version of the grammar. (Note that this pull request does not include specs for #39. My intention is to add specs for #39 on top of these baseline specs.)

I've actually never written CoffeeScript before and just relied on the example specs mentioned in #38 as reference. If anything looks wrong or improper let me know and I can make the necessary changes.

### CODE REVIEW

**Comment by 50Wliu** [comment]
The value and scope checks can be combined into a single line like so:
`expect(tokens[x]).toEqual value: "whatever", scopes: ["scopes"]`

For tests that involve quotes, you can switch the containing quotes so that you don't need to escape anything.  For example: `grammar.tokenizeLine '"foo bar"'`.

**Comment by bitbutt** [comment]
I've updated my branch to incorporate your suggestions. It's definitely more readable now. If there's any other adjustments you'd like I'd be happy to make them.

**Comment by bitbutt** [comment]
I've rebased my branch onto the current master.

File: spec/grammar-spec.coffee [hunk 1]
```diff
@@ -12,6 +12,181 @@ describe "Clojure grammar", ->
...
+    {tokens} = grammar.tokenizeLine "foo/bar"
+    expect(tokens[0]).toEqual value: "foo", scopes: ["source.clojure", "meta.symbol.namespace.clojure"]
+    expect(tokens[1]).toEqual value: "/", scopes: ["source.clojure"]
+    expect(tokens[2]).toEqual value: "bar", scopes: ["source.clojure", "meta.symbol.clojure"]
+
+  it "tokenizes trailing whitespace", ->
+    {tokens} = grammar.tokenizeLine "   \n"
+    expect(tokens[0]).toEqual value: "   \n", scopes: ["source.clojure", "invalid.trailing-whitespace"]
+
+  # TODO: Test 'after' tokens?
```
Comment (50Wliu) [inline spec/grammar-spec.coffee]: Is this something you want to finish in this PR?

**Comment by bitbutt** [comment]
Ah, yeah. I wasn't sure if it was worth testing the trailing whitespace tokens since they didn't necessarily have the same scopes across the different meta groups (and therefore didn't fit well into my generic function).

I'll leave it up to you whether I should (1) delete the todo and not worry about the after tokens, (2) add some testing of the tokens, or (3) do something else. Let me know and I can make the change.

**Comment by 50Wliu** [comment]
Sorry for the delay in replying.  Some specs are better than no specs, so let's remove the TODO for now and return to it later if necessary.

**Comment by bitbutt** [comment]
No worries! I've removed the TODO and squashed my commits.

**Comment by 50Wliu** [comment]
Thanks for this!

**Pull request merged**